src/test/java/htsjdk/samtools/cram/structure/BlockTest.java

```diff
@@ -0,0 +1,145 @@
+package htsjdk.samtools.cram.structure;
+
+import htsjdk.HtsjdkTest;
+import htsjdk.samtools.cram.CRAMException;
+import htsjdk.samtools.cram.common.CramVersions;
+import htsjdk.samtools.cram.common.Version;
+import htsjdk.samtools.cram.compression.ExternalCompressor;
+import htsjdk.samtools.cram.structure.block.*;
+import htsjdk.samtools.util.RuntimeIOException;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+
+public class BlockTest extends HtsjdkTest {
+
+    private void contentCheck(final Block actual,
+                             final byte[] expectedRaw,
+                             final byte[] expectedCompressed) {
+
+        // raw and compressed data are equal to what was given
+
+        Assert.assertEquals(actual.getUncompressedContent(), expectedRaw);
+        Assert.assertEquals(actual.getUncompressedContentSize(), expectedRaw.length);
+        Assert.assertEquals(actual.getCompressedContent(), expectedCompressed);
+        Assert.assertEquals(actual.getCompressedContentSize(), expectedCompressed.length);
+    }
+
+    @Test
+    public void uncompressedTest() {
+        final byte[] testData = "TEST STRING".getBytes();
+
+        final Block fhBlock = Block.createRawFileHeaderBlock(testData);
+        contentCheck(fhBlock, testData, testData);
+
+        final Block chBlock = Block.createRawCompressionHeaderBlock(testData);
+        contentCheck(chBlock, testData, testData);
+
+        final Block shBlock = Block.createRawSliceHeaderBlock(testData);
+        contentCheck(shBlock, testData, testData);
+
+        final Block core = Block.createRawCoreDataBlock(testData);
+        contentCheck(core, testData, testData);
+    }
+
+    private Block roundTrip(final Block in, final Version version) {
+        byte[] written;
+        try (final ByteArrayOutputStream os = new ByteArrayOutputStream()) {
+            in.write(version.major, os);
+            written = os.toByteArray();
+        } catch (final IOException e) {
```

Comment:
In tests we can just declare that these methods and the test methods "throw IOException" rather than catching and wrapping them, and just get rid of the catch clause.